packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/layout/layoutNode.ts

```diff
@@ -388,3 +388,16 @@ export const getWestBorderNodeFootprintHeight = (
 export const getChildren = (node: Node<NodeData>, allVisibleNodes: Node<NodeData>[]): Node<NodeData>[] => {
   return allVisibleNodes.filter((child) => child.parentNode === node.id);
 };
+
+export const applyRatioOnNewNodeSizeValue = (node: Node<NodeData>) => {
+  const initRatio = (node.data.defaultWidth ?? defaultWidth) / (node.data.defaultHeight ?? defaultHeight);
```

Comment:
Potential for initRatio to be "Infinity" here is `node.data.defaultHeight` returns 0.

Reply:
Done